pkg/armhelpers/azureclient.go

```diff
@@ -462,3 +463,47 @@ func (az *AzureClient) addAcceptLanguages() autorest.PrepareDecorator {
 		})
 	}
 }
+
+func (az *AzureClient) setAuxiliaryTokens() autorest.PrepareDecorator {
+	return func(p autorest.Preparer) autorest.Preparer {
+		return autorest.PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			r, err := p.Prepare(r)
+			if err != nil {
+				return r, err
+			}
+			if az.auxiliaryTokens != nil {
+				for _, token := range az.auxiliaryTokens {
+					if token == "" {
+						continue
+					}
+
+					r.Header.Set("x-ms-authorization-auxiliary", fmt.Sprintf("Bearer %s", token))
+				}
+			}
+			return r, nil
+		})
+	}
+}
+
+// AddAuxiliaryTokens sets the list of aux tokens to accept on this request
+func (az *AzureClient) AddAuxiliaryTokens(tokens []string) {
+	az.auxiliaryTokens = tokens
+	requestWithTokens := az.setAuxiliaryTokens()
+
+	az.authorizationClient.Client.RequestInspector = requestWithTokens
+	az.deploymentOperationsClient.Client.RequestInspector = requestWithTokens
+	az.deploymentsClient.Client.RequestInspector = requestWithTokens
+	az.deploymentsClient.Client.RequestInspector = requestWithTokens
```

Comment:
Line 496 and 495 are duplicates

Reply:
Thanks. Removed duplicates in both functions.